lib/core/src/recover/handlers/tests/handle_chain_receive_swap_tests_integration.rs

```diff
@@ -0,0 +1,601 @@
+                previous_output: bitcoin::OutPoint {
+                    txid: lockup_bitcon_tx_id,
+                    vout: 0,
+                },
+                script_sig: bitcoin::ScriptBuf::new(),
+                sequence: Sequence::default(),
+                witness: bitcoin::Witness::new(),
+            }],
+            output: vec![bitcoin::TxOut {
+                value: Amount::from_sat(amount - 1000),   // Subtract fee
+                script_pubkey: bitcoin::ScriptBuf::new(), // Destination address
+            }],
+        };
+
+        // Create history tx for refund
+        let refund_bitcoin_txid: Txid = refund_tx
+            .clone()
+            .compute_txid()
+            .to_string()
+            .parse()
+            .unwrap();
+        let refund_history_tx = HistoryTxId {
+            txid: refund_bitcoin_txid,
+            height: refund_height as i32,
+        };
+        // Add refund tx to script history
+        let mut script_history = context
+            .btc_script_to_history_map
+            .get(lockup_script)
+            .cloned()
+            .unwrap_or_default();
+
+        script_history.push(refund_history_tx);
+        context
+            .btc_script_to_history_map
+            .insert(lockup_script.clone(), script_history);
+
+        // Add refund tx to txs map
+        let mut txs = context
+            .btc_script_to_txs_map
+            .get(lockup_script)
+            .cloned()
+            .unwrap_or_default();
+        txs.push(refund_tx.clone());
+        context
+            .btc_script_to_txs_map
+            .insert(lockup_script.clone(), txs);
+
+        (context, lockup_tx_id, refund_tx.compute_txid().to_string())
+    }
+
+    // Helper to add LBTC transactions to the history
+    fn add_lbtc_history_to_context(
+        mut context: RecoveryContext,
+        claim_script: &elements::Script,
+        tx_ids: &[(/*tx_id*/ &str, /*height*/ u32)],
+        claim_tx_id_hex: &str,
+        amount: u64,
+    ) -> RecoveryContext {
+        // Add history txs
+        let mut history = Vec::new();
+        for (tx_id_hex, height) in tx_ids {
+            let tx_id = elements::Txid::from_str(tx_id_hex).unwrap();
+            history.push(HistoryTxId {
+                txid: tx_id,
+                height: *height as i32,
+            });
+
+            // If this is the claim tx, add it to the incoming tx map
+            if *tx_id_hex == claim_tx_id_hex {
+                let wallet_tx = create_mock_lbtc_wallet_tx(tx_id_hex, *height, amount as i64);
+                context.tx_map.incoming_tx_map.insert(tx_id, wallet_tx);
+            }
+        }
+
+        // Add to history map
+        context
+            .lbtc_script_to_history_map
+            .insert(claim_script.clone(), history);
+
+        context
+    }
+
+    // Create a mock LBTC wallet transaction
+    fn create_mock_lbtc_wallet_tx(tx_id_hex: &str, height: u32, amount: i64) -> WalletTx {
```

Comment:
Seems like this fn and similar `create_mock_wallet_tx` fns are repeated in other files, could be added to mod.rs

Reply:
Moved all shared methods.